src/client/sandbox/cookie/window-sync.ts

```diff
@@ -165,12 +165,12 @@ export default class WindowSync {
         }
 
         if (syncMessages.length) {
-            let syncMessagesPromise = Promise.all(syncMessages);
+            const syncMessagesPromise = Promise.all(syncMessages);
 
             if (this._win === topOpenerWindow)
-                syncMessagesPromise = syncMessagesPromise.then(() => this._removeSyncCookie(cookies));
+                return syncMessagesPromise.then(() => this._removeSyncCookie(cookies));
 
-            return syncMessagesPromise;
+            return syncMessagesPromise.then();
```

Comment:
Why do we need this `.then()`?

Reply:
Because of the return type of this method. It's declared as `Promise<void>`, but the `syncMessagesPromise` variable type is `Promise<void[]>` (because of `Promise.all` usage).